pkg/ibmvpc/vpc.go

```diff
@@ -755,6 +790,103 @@ func (tgw *TransitGateway) AllowedConnectivity(src, dst vpcmodel.VPCResourceIntf
 	return nil, errors.New("TransitGateway.AllowedConnectivity() expected src and dst to be two nodes or two subnets")
 }
 
+func (tgw *TransitGateway) RouterDefined(src, dst vpcmodel.Node) bool {
+	// destination node has a transit gateway connection iff a prefix (possibly default) is defined for it
+	dstNodeHasTgw := tgw.prefixOfSrcDst(src, dst) != nil
+	if vpcmodel.HasNode(tgw.sourceNodes, src) && dstNodeHasTgw {
+		return true
+	}
+	return false
+}
+
+// gets a string description of prefix indexed "index" from TransitGateway tgw
+func prefixDefaultStr(tc *datamodel.TransitConnection) (string, error) {
+	actionName, err := actionNameStr(tc.PrefixFiltersDefault)
+	if err != nil {
+		return "", err
+	}
+	return fmt.Sprintf(" default prefix,  action: %s", actionName), nil
+}
+
+func (tgw *TransitGateway) tgwPrefixStr(prefix tgwPrefix) (string, error) {
+	// Array of prefix route filters for a transit gateway connection. This is order dependent with those first in the
+	// array being applied first, and those at the end of the array is applied last, or just before the default.
+	resStr := fmt.Sprintf("transit-connection: %s", *prefix.tc.Name)
+	if prefix.index == -1 { // default
+		defaultStr, err := prefixDefaultStr(prefix.tc)
+		if err != nil {
+			return "", err
+		}
+		return resStr + defaultStr, nil
+	}
+	if len(prefix.tc.PrefixFilters) < prefix.index+1 {
+		return "", fmt.Errorf("np-guard error: prefix index %d does not exists in transit connection %s of transit gateway %s",
+			prefix.index, *prefix.tc.Name, tgw.Name())
+	}
+	prefixFilter := prefix.tc.PrefixFilters[prefix.index]
+	actionName, err := actionNameStr(prefixFilter.Action)
+	if err != nil {
+		return "", err
+	}
+	resStr += fmt.Sprintf(", index: %v, action: %s", prefix.index, actionName)
+	if prefixFilter.Ge != nil {
+		resStr += fmt.Sprintf(", ge: %v", *prefixFilter.Ge)
+	}
+	if prefixFilter.Le != nil {
+		resStr += fmt.Sprintf(", le: %v", *prefixFilter.Le)
+	}
+	resStr += fmt.Sprintf(", prefix: %s", *prefixFilter.Prefix)
+	return resStr, nil
+}
+
+// for an action of type *string as stored in *datamodel.TransitConnection returns allow/deny
+func actionNameStr(action *string) (string, error) {
+	actionBool, err := parseActionString(action)
+	if err != nil {
+		return "", err
+	}
+	if actionBool {
+		return "allow", nil
+	}
+	return "deny", nil
+}
```

Comment:
done. note that it effects the printing - "permit" instead of "allow" which is non-consistent with the acl and sg wording in the same printing.
Perhaps we should consider making it consistent by changing permit to allow?

Reply:
I'd keep it , as these are the terms the user is familiar with..
@zivnevo @kyorav  any other thoughts here?

Reply:
I agree with Adi, it is best to be consistent with the terms used by IBM Cloud